luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -106,8 +125,12 @@ local function getImpulseMultiplier(unitDefID, weaponDefID, damage)
 		impulseFactor = weaponDefIDImpulses[weaponDefID].impulseFactor or 1
 	end
 	local impulse = (damage + impulseBoost) * impulseFactor
-	local maxImpulse = unitsMaxImpulse[unitDefID]
-	local impulseMultiplier = mathMin(maxImpulse/impulse, 1) -- negative impulse values are not capped.
+	local impulseMultiplier
+	if impulse < unitsMinImpulse[unitDefID] then
+		impulseMultiplier = 0
+	else
+		impulseMultiplier = mathMin(unitsMaxImpulse[unitDefID]/impulse, 1) -- negative impulse values are not capped.
```

Comment:
Can `impulse` ever be 0?

Reply:
good catch, added a bit to impulseboost by default to prevent this error.

Reply:
Why? If impulse is 0, just return the default value of impulse multiplier, since anything times 0 is 0.

Reply:
actually, this is taken care of already.

line 129 checksif impulse < unitsMinImpulse[unitDefID], which if it returns true sets impulseMultiplier to 0, then the else right after that performs the impulse calculation with division operation at which point it can't be 0.

Reply:
reverted my change, because not necessary